examples/interactions/src/jvmMain/kotlin/ReconnectOnConnectFailExample.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.rsocket.kotlin.*
+import io.rsocket.kotlin.core.*
+import io.rsocket.kotlin.payload.*
+import io.rsocket.kotlin.transport.*
+import io.rsocket.kotlin.transport.local.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.*
+
+@TransportApi
+fun main(): Unit = runBlocking {
+    val server = LocalServer()
+    RSocketServer().bind(server) {
+        RSocketRequestHandler {
+            requestStream {
+                val data = it.data.readText()
+                val metadata = it.metadata?.readText()
+                println("Server received payload: data=$data, metadata=$metadata")
+
+                flow {
+                    repeat(50) { i ->
+                        delay(100)
+                        emit(Payload("Payload: $i", metadata))
+                        println("Server sent: $i")
+                    }
+                }
+            }
+        }
+    }
+
+    //emulate connection establishment error
+    var first = true
+
+    val rSocket = RSocketConnector {
+        //reconnect 10 times with 1 second delay if connection establishment failed
```

Comment:
As an example I think a goal here could be to make this comment redundant throught he expressiveness of the method call.

```
reconnectable(retries = 10, delay = 1.seconds, predicate = ALWAYS)
```

Likewise with the println, how do we show these events without sprinkling reconnect in the code? Not against it here, but when I'm debugging rsocket-kotlin being able to turn on either frame logging or transport event logging would be so helpful.

Reply:
regarding first point - upper
regarding second - yes, will add debug logging for that